File: Project/Boundaries/Rule.py


class Rule:
    def __init__(self, rule_id: str, rule_name: str, rule_type: str, rule_data: str):
        self.id = str(rule_id)
        self.ruleName = rule_name
        self.ruleType = rule_type
        self.ruleData = rule_data

    def __str__(self):
        return f"RuleId:{self.id}, " \
               f"RuleName:{self.ruleName}, " \
               f"RuleType:{self.ruleType}, " \
               f"RuleData:{self.ruleData}"

File: Web/src/Rules/RulesClasses/Rule.js
import { makeObservable, observable } from 'mobx';

export default class Rule {
    id = 0;
    name = '';
    type = '';
    value = '';

    constructor({ id, ruleName, ruleType, ruleData }) {
        makeObservable(this, {
            id: observable,
            name: observable,
            type: observable,
            value: observable,
        });
        this.id = id;
        this.name = ruleName;
        this.type = ruleType;
        this.value = ruleData;
    }

    /**
     *
     * @param {string} name
     * @param {string} type
     * @param {string} data
     */
    updateRule(ruleName, ruleType, ruleData) {
        this.name = ruleName;
        this.type = ruleType;
        this.value = ruleData;
    }
}
